ccm_web/client/src/pages/AddNonUMUsers.tsx

```diff
@@ -0,0 +1,163 @@
+      }
+      tryAgain={clearGetSectionsError}
+    />
+  )
+
+  const renderSelectInputMethod = (): JSX.Element => {
+    if (getSectionsError !== undefined) return getSectionsErrorAlert
+
+    return (
+      <div className={classes.container}>
+        <MethodSelect<InputMethod>
+          label='Choose how you want to add users'
+          options={[
+            { key: 'single', label: 'Add one user manually' },
+            { key: 'csv', label: 'Add multiple users by uploading a CSV' }
+          ]}
+          typeGuard={(v): v is InputMethod => v === 'single' || v === 'csv'}
+          selectedMethod={inputMethod}
+          setMethod={setInputMethod}
+          disabled={isGetSectionsLoading}
+          onButtonClick={() => {
+            if (inputMethod === 'csv') {
+              setActivePageState(PageState.AddCSVUsers)
+            } else {
+              setActivePageState(PageState.AddSingleUser)
+            }
+          }}
+        />
+        <Backdrop className={classes.backdrop} open={isGetSectionsLoading}>
+          <Grid container>
+            <Grid item xs={12}>
+              <CircularProgress color='inherit' />
+            </Grid>
+            <Grid item xs={12}>
+              Loading section data from Canvas
+            </Grid>
+          </Grid>
+        </Backdrop>
+      </div>
+    )
+  }
+
+  const resetFeature = (): void => setActivePageState(PageState.SelectInputMethod)
+
+  const renderActivePageState = (state: PageState): JSX.Element => {
+    switch (state) {
+      case PageState.SelectInputMethod:
+        return renderSelectInputMethod()
+      case PageState.AddSingleUser:
+        return (
+          <UserEnrollmentForm
+            sections={sections ?? []}
+            rolesUserCanEnroll={rolesUserCanEnroll}
+            resetFeature={resetFeature}
+            settingsURL={settingsURL}
+          />
+        )
+      case PageState.AddCSVUsers:
+        return (
+          <MultipleUserEnrollmentWorkflow
+            course={props.course}
+            sections={sections ?? []}
+            onSectionCreated={
+              (newSection) => {
+                if (sections !== undefined) {
+                  setSections(
+                    sortSections(sections.concat(injectCourseName([newSection], props.course.name)))
+                  )
+                }
+              }
+            }
+            rolesUserCanEnroll={rolesUserCanEnroll}
+            resetFeature={resetFeature}
+            settingsURL={settingsURL}
+          />
+        )
+      default:
+        return <ErrorAlert />
+    }
+  }
+
+  return (
+    <div className={classes.root} aria-live='polite'>
+      <Help baseHelpURL={props.globals.baseHelpURL} helpURLEnding={props.helpURLEnding} />
+      <Typography variant='h5' component='h1' className={classes.spacing}>{props.title}</Typography>
+      {renderActivePageState(activePageState)}
+    </div>
+  )
+}
```

Comment:
The code in this file is very well organized, easy to read and separating each selection to a component is really neat.